### PR TITLE
Add management of Meadow.Cloud API keys

### PR DESCRIPTION
**Pull request by stevenkuhn**
This adds the ability to manage your Meadow.Cloud API keys via the command line (V2 only).

The commands are:
```
meadow cloud apikey create <NAME> --duration|d <duration> --scopes|s <scopes>
meadow cloud apikey list
meadow cloud apikey update <NAME_OR_ID> --name|n <name> --scopes|s <scopes>
meadow cloud apikey delete <NAME_OR_ID>
```

This also utilizes the new Console spinner and implements a new `ConsoleTable` for displaying tabular data.

### Create

```
> meadow cloud api create myapikey --duration 30 --scopes command
Creating an API key on Meadow.Cloud...
Your API key 'myapikey' (expiring 1/26/2024 5:40:30 PM UTC) is:

mc_xxxxxxxxxxxxxxxxxxxxxxxxxxx_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

Make sure to copy this key now as you will not be able to see this again.
Done.
```

### Update

```
❯ meadow cloud apikey update myapikey --scopes command search
Updating API key `myapikey` on Meadow.Cloud...
Done.
```

### Delete

```
❯ meadow cloud apikey delete myapikey
Deleting API key `myapikey` on Meadow.Cloud...
Done.
```

#### When you have no API keys

```
❯ meadow cloud apikey list
Retrieving your API keys from Meadow.Cloud...
You have no API keys.
Done.
```

#### When you have a single API key

```
❯ meadow cloud apikey list
Retrieving your API keys from Meadow.Cloud...

---------------------------------+----------+----------------------+----------------
Id                               | Name     | Expires (UTC)        | Scopes
---------------------------------|----------|----------------------|----------------
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx | myapikey | 1/26/2024 5:44:33 PM | command, search
---------------------------------+----------+----------------------+----------------

Done.
```

### CODE REVIEW

**Comment by adrianstevens** [comment]
This PR looks great except we'll be removing the `WithSpinner` extension method.

**Comment by stevenkuhn** [comment]
> This PR looks great except we'll be removing the `WithSpinner` extension method.

Will the spinner be removed entirely or replaced with something else? Or is only the extension method going away?

**Comment by CartBlanche** [comment]
@adrianstevens The WithSpinner calls are't in the base classes, so it isn't mixing UI with functionality. It is specifically for the console UI, unless I've missed something.

@stevenkuhn due to me messing up the `develop` branch you may have to resolve some merge conflicts with this new `develop` branch.

**Comment by adrianstevens** [comment]
Let's definitely talk about the spinner code but adding extension methods to tasks to trigger presentation is basically the opposite of separation-of-concerns. And if extinction methods on tasks are needed to give feedback to the user, it's a pretty good sign that we have some architecture issues to solve.

**Comment by stevenkuhn** [comment]
I updated this PR to branch off of and merge back into `v2`. I also removed the calls to the spinner

**Comment by stevenkuhn** [comment]
Updated my branch with the latest in `develop` and I removed the `WithSpinner` calls.

**Comment by stevenkuhn** [comment]
@lamebrain If you want to test with the branch, you can run the command like this:

```
> dotnet run --framework net6.0 -- cloud apikey <command>
```

For example, to create an API key in production, it would be:

```
> dotnet run --framework net6.0 -- cloud api create myapikey --duration 30 --scopes command
```

File: Source/v2/Meadow.CLI/Commands/Current/Cloud/ApiKey/CloudApiKeyCreateCommand.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+using Meadow.Cloud.Identity;
+using Microsoft.Extensions.Logging;
+
+namespace Meadow.CLI.Commands.DeviceManagement;
+
+[Command("cloud apikey create", Description = "Create a Meadow.Cloud API key")]
+public class CloudApiKeyCreateCommand : BaseCloudCommand<CloudApiKeyCreateCommand>
+{
+    [CommandParameter(0, Description = "The name of the API key", IsRequired = true, Name = "NAME")]
+    public string? Name { get; set; }
```
Comment (adrianstevens) [inline Source/v2/Meadow.CLI/Commands/Current/Cloud/ApiKey/CloudApiKeyCreateCommand.cs:13]: if this is required it doesn't need to be nullable

File: Source/v2/Meadow.CLI/Commands/Current/Cloud/ApiKey/CloudApiKeyCreateCommand.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    [CommandOption("duration", 'd', Description = "The duration of the API key, in days", IsRequired = true)]
+    public int Duration { get; set; }
+
+    [CommandOption("scopes", 's', Description = "The list of scopes (permissions) to grant the API key", IsRequired = true)]
+    public string[]? Scopes { get; set; }
```
Comment (adrianstevens) [inline Source/v2/Meadow.CLI/Commands/Current/Cloud/ApiKey/CloudApiKeyCreateCommand.cs:19]: And here

File: Source/v2/Meadow.CLI/Commands/Current/Cloud/ApiKey/CloudApiKeyUpdateCommand.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+using Meadow.Cloud;
+using Meadow.Cloud.Identity;
+using Microsoft.Extensions.Logging;
+
+namespace Meadow.CLI.Commands.DeviceManagement;
+
+[Command("cloud apikey update", Description = "Update a Meadow.Cloud API key")]
+public class CloudApiKeyUpdateCommand : BaseCloudCommand<CloudApiKeyUpdateCommand>
+{
+    [CommandParameter(0, Description = "The name or ID of the API key", IsRequired = true, Name = "NAME_OR_ID")]
```
Comment (adrianstevens) [inline Source/v2/Meadow.CLI/Commands/Current/Cloud/ApiKey/CloudApiKeyUpdateCommand.cs:12]: nullable

**Review by adrianstevens — APPROVED** [review]

**Pull request merged**